src/goto-programs/goto_sideeffects.cpp

```diff
@@ -675,13 +675,13 @@ void goto_convertt::remove_function_call(
 
     new_base_name += '_';
     new_base_name += id2string(symbol.name);
-    new_base_name += "$" + std::to_string(++temporary_counter);
+    new_base_name += "$" + std::to_string(++tmp_symbol.counter);
```

Comment:
do we need to manually increment the counter here? Shouldn't we be calling new_symbol?

Reply:
I didn't want to change semantics of code other than that touching compound literals. If we want to use new_symbol() here (which sounds like a good idea), we'd need to take a careful look at `goto_convertt::new_name()` and how to possibly merge it with `new_symbol()`. They are different: `new_symbol()` sets the `lvalue` attribute, which I'm not sure this code path here expects.